tests/parser/syntax/test_interfaces.py

```diff
@@ -0,0 +1,133 @@
+import pytest
+
+from vyper import (
+    compiler,
+)
+from vyper.exceptions import (
+    InvalidTypeException,
+    StructureException,
+    TypeMismatchException,
+    VariableDeclarationException,
+)
+
+fail_list = [
+    ("""
+from vyper.interfaces import ERC20
+a: public(ERC20)
+@public
+def test():
+    b: uint256 = self.a
+    """, TypeMismatchException),
+    ("""
+from vyper.interfaces import ERC20
+aba: public(ERC20)
+@public
+def test():
+    self.aba = ERC20
+    """, VariableDeclarationException),
+    ("""
+from vyper.interfaces import ERC20
+
+a: address(ERC20) # invalid syntax now.
+    """, StructureException),
+    ("""
+from vyper.interfaces import ERC20
+
+@public
+def test():
+    a: address(ERC20) = ZERO_ADDRESS
+    """, InvalidTypeException),
+    ("""
+a: address
+
+@public
+def test():  # may not call normal address
+    assert self.a.random()
+    """, StructureException),
+    ("""
+from vyper.interfaces import ERC20
+@public
+def test(a: address):
+    my_address: address = ERC20()
+    """, TypeMismatchException)
+]
+
+
+@pytest.mark.parametrize('bad_code', fail_list)
+def test_interfaces_fail(bad_code):
+    with pytest.raises(bad_code[1]):
+        compiler.compile_code(bad_code[0])
+
+
+valid_list = [
+    """
+from vyper.interfaces import ERC20
+b: ERC20
+@public
+def test(input: address):
+    assert self.b.totalSupply() == ERC20(input).totalSupply()
+    """,
+    """
+from vyper.interfaces import ERC20
+
+contract Factory:
+   def getExchange(token_addr: address) -> address: constant
```

Comment:
Can you do a `getToken(exchange_addr: address) -> address`  case that does the opposite of `test()`?

Reply:
@fubuloubu Opposite in what way ?

Reply:
Test that you can get exchange address for the token (`ERC20` is an internal interface, `Factory` is defined in code), then test that you can get the token for a given exchange.